src/Modules/OrchardCore.Commerce/Controllers/PaymentController.cs

```diff
@@ -157,40 +159,42 @@ public async Task<IActionResult> Validate()
         }
     }
 
-    [Route("PaymentRequest/{orderId}")]
+    [Route("checkout/paymentrequest/{orderId}")]
     public async Task<IActionResult> PaymentRequest(string orderId)
     {
         if (await _contentManager.GetAsync(orderId) is not { } order) return NotFound();
 
-        // Users should only see their own Orders.
-        if (order.Author != User.Identity.Name)
+        if (string.IsNullOrEmpty(User.Identity.Name))
         {
-            return NotFound();
+            return LocalRedirect($"~/Login?ReturnUrl=~/Contents/ContentItems/{orderId}");
         }
 
         var orderPart = order.As<OrderPart>();
 
-        // If status is not Pending or there are no line items, there is nothing to be done.
-        if (!string.Equals(orderPart.Status.Text, OrderStatuses.Pending, StringComparison.OrdinalIgnoreCase) ||
-            !orderPart.LineItems.Any())
+        // If there are no line items, there is nothing to be done.
+        if (!orderPart.LineItems.Any())
         {
-            return NotFound();
+            await _notifier.InformationAsync(H["This Order contains no line items, so there is nothing to be paid."]);
+            return LocalRedirect($"~/Contents/ContentItems/{orderId}");
```

Comment:
Use the `this.RedirectToContentDisplay(orderId)` extension method everywhere. It's both cleaner and returns the correct URL even if autoroute part or similar is used.